plotutil/xyer.go

```diff
@@ -0,0 +1,77 @@
+package plotutil
+
+import "github.com/gonum/plot/plotter"
+
+type xyer struct {
+	x, y []float64
+	len  int
+}
+
+// Len returns the length of the XYer object.
+// fulfills the plotter.XYer interface requirement.
+func (xy *xyer) Len() int {
+	return xy.len
+}
+
+// XY returns the x,y values at index idx.
+// fulfills the plotter.XYer interface requirement.
+func (xy *xyer) XY(idx int) (x, y float64) {
+	return xy.x[idx], xy.y[idx]
+}
+
+// ZipXY is a convenience function to build a plotter.XYer from slices.
+// Length will be set to the shortest slice length.
+// Panics on empty or nil slices.
+func ZipXY(x, y []float64) plotter.XYer {
```

Comment:
@Kunde21  I don't see any reason we would change the implementation in a meaningful way, so I don't think there's any need to hide the implementation behind an interface. The explicit approach I suggested is simpler, more transparent, and makes it obvious that the data is not copied.

As to shortening the length automatically, that's against the spirit of gonum APIs. Inputs are either correct or we panic. We don't silently make corrections to make an output happen; that's the easiest way to have bugs which are never discovered.